### ISSUE TITLE
[BUG] collection_ops_test.py::test_sequence_too_long_sequence[Long(not_null)][DATAGEN_SEED=1701553915, INJECT_OOM]

### ISSUE THREAD

**Issue by pxLi** [NVIDIA/spark-rapids#9933]
**Describe the bug**
rapids_integration-dev-github, build ID 866

failed
```
FAILED ../../src/main/python/collection_ops_test.py::test_sequence_too_long_sequence[Long(not_null)][DATAGEN_SEED=1701553915, INJECT_OOM] - Failed: DID NOT RAISE <class 'Exception'>
```

```

[2023-12-02T23:20:46.759Z] _______________ test_sequence_too_long_sequence[Long(not_null)] ________________

[2023-12-02T23:20:46.759Z] [gw4] linux -- Python 3.9.18 /opt/conda/bin/python

[2023-12-02T23:20:46.759Z] 

[2023-12-02T23:20:46.759Z] stop_gen = Long(not_null)

[2023-12-02T23:20:46.759Z] 

[2023-12-02T23:20:46.759Z]     @pytest.mark.parametrize('stop_gen', sequence_too_long_length_gens, ids=idfn)

[2023-12-02T23:20:46.759Z]     @allow_non_gpu(*non_utc_allow)

[2023-12-02T23:20:46.759Z]     def test_sequence_too_long_sequence(stop_gen):

[2023-12-02T23:20:46.759Z] >       assert_gpu_and_cpu_error(

[2023-12-02T23:20:46.759Z]             # To avoid OOM, reduce the row number to 1, it is enough to verify this case.

[2023-12-02T23:20:46.759Z]             lambda spark:unary_op_df(spark, stop_gen, 1).selectExpr(

[2023-12-02T23:20:46.759Z]                 "sequence(0, a)").collect(),

[2023-12-02T23:20:46.759Z]             conf = {}, error_message = "Too long sequence")

[2023-12-02T23:20:46.759Z] 

[2023-12-02T23:20:46.759Z] ../../src/main/python/collection_ops_test.py:329: 

[2023-12-02T23:20:46.759Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2023-12-02T23:20:46.759Z] ../../src/main/python/asserts.py:646: in assert_gpu_and_cpu_error

[2023-12-02T23:20:46.759Z]     assert_spark_exception(lambda: with_cpu_session(df_fun, conf), error_message)

[2023-12-02T23:20:46.759Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2023-12-02T23:20:46.759Z] 

[2023-12-02T23:20:46.759Z] func = <function assert_gpu_and_cpu_error.<locals>.<lambda> at 0x7fda5085d5e0>

[2023-12-02T23:20:46.759Z] error_message = 'Too long sequence'

[2023-12-02T23:20:46.759Z] 

[2023-12-02T23:20:46.759Z]     def assert_spark_exception(func, error_message):

[2023-12-02T23:20:46.759Z]         """

[2023-12-02T23:20:46.759Z]         Assert that a specific Java exception is thrown

[2023-12-02T23:20:46.759Z]         :param func: a function to be verified

[2023-12-02T23:20:46.759Z]         :param error_message: a string such as the one produce by java.lang.Exception.toString

[2023-12-02T23:20:46.759Z]         :return: Assertion failure if no exception matching error_message has occurred.

[2023-12-02T23:20:46.759Z]         """

[2023-12-02T23:20:46.759Z]         with pytest.raises(Exception) as excinfo:

[2023-12-02T23:20:46.760Z] >           func()

[2023-12-02T23:20:46.760Z] E           Failed: DID NOT RAISE <class 'Exception'>
```

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

**Expected behavior**

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**

**Comment by revans2** [comment]
We just hit this again with seed DATAGEN_SEED=1701786279

**Comment by jlowe** [comment]
Was supposed to be fixed by #9865 but there is a bug in LongGen that will generate special cases even when none are requested.  That fix is posted at #9960 and will fix this issue.
